client/state/jetpack-connect/actions.js

```diff
@@ -348,23 +351,24 @@ export default {
 				} );
 				// Update the user now that we are fully connected.
 				userFactory().fetch();
-				return wpcom.me().sites( { site_visibility: 'all', include_domain_only: true } );
+				dispatch( {
+					type: SITE_REQUEST,
```

Comment:
We're dispatching a `SITE_REQUEST`, indicating that a site request has started, and a `SITE_REQUEST_SUCCESS` upon successful request. However, if the request fails, we're not dispatching a `SITE_REQUEST_FAILURE`, so if the request fails for some reason, we'll not get notified that the request has finished; and a subsequent request to the same site might not be handled properly because, we'll incorrectly assume that a site request is currently in process.